### PR TITLE
feat: svelte 4 support

### PR DESCRIPTION
**Pull request by sajaghori**
* fixes the derived context error, updates the dependencies
* adds support for svelte 4

### CODE REVIEW

File: src/lib/context.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { getContext, setContext } from 'svelte'
-import { noop } from 'svelte/internal'
```
Comment (ivanhofer) [inline src/lib/context.ts:2]: is this no longer exported?

**Review by ivanhofer — COMMENTED** [review]
Hi @sajaghori, thanks for the PR. Can you please tell me what error you try to fix?

File: src/lib/context.ts (hunk 1 above)
Comment (sajaghori) [reply src/lib/context.ts:2]: exactly

File: src/lib/context.ts [hunk 2]
```diff
@@ -48,7 +48,12 @@ export const createSthemerContext = (strategy: SthemerStrategy = 'auto') => {
 
 	const strategyStore = writable<SthemerStrategy>(strategy)
 
-	const parentContext = getContext<SthemerContext>(KEY)
+	const parentContext =
+		getContext<SthemerContext>(KEY) ??
+		setContext<SthemerContext>(KEY, {
```
[the remark this answers is not in the document]
Comment (sajaghori) [reply src/lib/context.ts:53]: The error is: `Error: derived() expects stores as input, got a falsy value`

[the remark this answers is not in the document]
Comment (sajaghori) [reply src/lib/context.ts:53]: here is the change: https://github.com/sveltejs/svelte/commit/ea73930132b8185bed670409cc4fb245f44e78ad#diff-450b07486f98abe0cc46759ca76962d8c0fbc558c9c9648fe6ea15699915e195

[the remark this answers is not in the document]
Comment (ivanhofer) [reply src/lib/context.ts:53]: Ah got it. Please remove the `setContext` call. You can set the fallback value without storing it to the context. 
You can also get rid of the `?` in line 63.

[the remark this answers is not in the document]
Comment (sajaghori) [reply src/lib/context.ts]: yes, good idea!

**Review by ivanhofer — APPROVED** [review]

**Comment by ivanhofer** [comment]
Thanks @sajaghori

**Pull request merged**